### PR TITLE
[CI Environment] Several pipeline fixes, updated naming & minor improvements

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated step names in GH workflows
- Fixed incorrect Azure DevOps pipelines
- Improved test file fetching

| Pipeline |
| - |
| |

# Type of Change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: utilities/pipelines/sharedScripts/Get-ModuleTestFileList.ps1 [hunk 1]
```diff
@@ -35,7 +35,7 @@ function Get-ModuleTestFileList {
 
     $deploymentTests = @()
     if (Test-Path (Join-Path $ModulePath $SearchFolder)) {
-        $deploymentTests += (Get-ChildItem -Path (Join-Path $ModulePath $SearchFolder) -Depth 0 -Include ('*.json', '*.bicep') -File).FullName
+        $deploymentTests += (Get-ChildItem -Path (Join-Path $ModulePath $SearchFolder) -Recurse -Include ('*.json', 'deploy.test.bicep') -File).FullName
```
Comment (ahmadabdalla) [inline utilities/pipelines/sharedScripts/Get-ModuleTestFileList.ps1:38]: Why not make `('*.json', 'deploy.test.bicep')` as an array parameter with default values?

Comment (MrMCake) [reply utilities/pipelines/sharedScripts/Get-ModuleTestFileList.ps1:38]: We could. Just happens that we use this implementation in different locations. Will update it for this script :)

Comment (MrMCake) [reply utilities/pipelines/sharedScripts/Get-ModuleTestFileList.ps1]: done

**Review by ahmadabdalla — APPROVED** [review]
Approved 😇😇

**Pull request merged**